### PR TITLE
fix: Use idp client id from remote (WPB-6494)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6494" title="WPB-6494" target="_blank">BugWPB-6494</a>  [Android] Use IDP clientId from remote
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

ClientID is hardcoded 

### Solutions

Get client id from url and remove client secret

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3050](https://10.10.124.134/job/AR-build-pipeline/3050/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8876864)(8876864)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8876864)(8876864)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/feature/e2ei/OAuthUseCase.kt [hunk 1]
```diff
@@ -38,99 +39,90 @@ import net.openid.appauth.AuthorizationResponse
...
 
-    fun launch(activityResultRegistry: ActivityResultRegistry, resultHandler: (OAuthResult) -> Unit) {
+    fun launch(
+        activityResultRegistry: ActivityResultRegistry,
+        resultHandler: (OAuthResult) -> Unit
+    ) {
         authState.performActionWithFreshTokens(authorizationService) { _, idToken, exception ->
             if (exception != null) {
-                Log.e("OAuthTokenRefreshManager", "Error refreshing tokens, continue with login!", exception)
+                Log.e(
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/feature/e2ei/OAuthUseCase.kt]: suggestion: use `appLogger`

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/util/UriUtil.kt [hunk 2]
```diff
@@ -66,3 +66,12 @@ fun URI.removeQueryParams(): URI {
...
 }
+
+fun URI.findParameterValue(parameterName: String): String? {
+    return rawQuery.split('&').map {
+        val parts = it.split('=')
+        val name = parts.firstOrNull() ?: ""
+        val value = parts.drop(1).firstOrNull() ?: ""
+        Pair(name, value)
+    }.firstOrNull { it.first == parameterName }?.second
+}
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/util/UriUtil.kt]: Could you please cover this with tests?

**Comment by AndroidBob** [comment]
Build [3076](https://10.10.124.134/job/AR-build-pipeline/3076/) **failed**.

**Comment by AndroidBob** [comment]
Build [3086](https://10.10.124.134/job/AR-build-pipeline/3086/) **failed**.

**Comment by AndroidBob** [comment]
Build [3098](https://10.10.124.134/job/AR-build-pipeline/3098/) **failed**.

**Comment by AndroidBob** [comment]
Build [3100](https://10.10.124.134/job/AR-build-pipeline/3100/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8886672)(8886672)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8886672)(8886672)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/feature/e2ei/OAuthUseCase.kt [hunk 3]
```diff
@@ -143,7 +133,8 @@ class OAuthUseCase(context: Context, private val authUrl: String, private val cl
 
     private fun handleAuthorizationResponse(intent: Intent, resultHandler: (OAuthResult) -> Unit) {
         val authorizationResponse: AuthorizationResponse? = AuthorizationResponse.fromIntent(intent)
-        val clientAuth: ClientAuthentication = ClientSecretBasic(CLIENT_SECRET)
+//        val clientAuth: ClientAuthentication = ClientSecretBasic(CLIENT_SECRET)
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/feature/e2ei/OAuthUseCase.kt]: Leftover?

**Review by mchenani — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3108](https://10.10.124.134/job/AR-build-pipeline/3108/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8887340)(8887340)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8887340)(8887340)-staging-compat.apk)

**Pull request merged**